### DISCUSSION TITLE
Using Solr with Docker

### DISCUSSION THREAD

**Discussion by Git-Romer** [irgroup-classrooms/dis17-2021#19] [Show and tell]
**German version below**

**Tutorial can be found [here](https://github.com/irgroup-classrooms/dis17-2021/blob/main/Solr-Docker/README.md)**

Hi,
I've seen that some people have a hard time installing or executing commands with Solr. Mostly it was due to dependency problems.
In this sense I would like to **offer you my help** if you still have no success and would like to install Solr in a "virtual environment".
I myself use a Docker container where **all dependencies are already installed** automatically and which once you have everything set up has a minimal maintenance effort. This also **keeps the local machine clean and safe** from temporary files or possible conflicts between individual programs/services.
I also thought about writing a tutorial about it but finally decided against it because it would have been quite a big effort to design the whole thing foolproof (also there is plenty of good tutorials out there already. I used a couple myself to get it to work). 
So **whoever wants to can contact me** and I would then go step by step through the installation with him/her.

Hi,
Habe gesehen, dass einige Leute Schwierigkeiten haben, Solr zu installieren oder Befehle mit Solr auszuführen. Meistens lag es an Abhängigkeits-Problemen.
In diesem Sinne möchte ich euch meine Hilfe anbieten, falls ihr noch keinen Erfolg hattet und Solr in einer "virtuellen Umgebung" installieren möchtet.
Ich selbst verwende einen Docker-Container, bei dem alle Abhängigkeiten bereits automatisch installiert sind und der, wenn man alles eingerichtet hat, einen minimalen Wartungsaufwand hat. So bleibt auch der lokale Rechner sauber und sicher vor temporären Dateien oder möglichen Konflikten zwischen einzelnen Programmen/Diensten.
Ich habe auch darüber nachgedacht, ein Tutorial darüber zu schreiben, mich aber schließlich dagegen entschieden, weil es ein ziemlicher Aufwand gewesen wäre, das Ganze möglichst foolproof zu gestalten (außerdem gibt es schon jede Menge gute Tutorials da draußen. Ich habe selbst ein paar genutzt, um alles zum Laufen zu bringen).
Wer also bedarf/Interesse hat, kann sich bei mir melden und ich würde dann Schritt für Schritt mit der/demjenigen die Installation durchgehen.

**Comment by phschaer** [comment]
While writing your own tutorial might be "overkill", you can think of screen grabbing the basic steps. I can offer to host and link the video if this would help the rest of the group.

**Reply by Git-Romer** [reply]
Sure. If this is something that people can take advantage of. Might want to discuss this in a zoom session?!

**Reply by Git-Romer** [reply]
[Pull request](https://github.com/irgroup-classrooms/dis17-2021/pull/20) has been created.
Will update my original post and add the link as soon as the pull request has been merged ;)

**Reply by phschaer** [reply]
I just merged your pull request. Thank you very much for this additional piece of advice and help for the rest of the group. Did anybody tested/reproduced this tutorial so far? (and not just "read it", like I did *scnr*)

**Reply by Git-Romer** [reply]
Perfect Thank you!
Mr. Pekarski went through the tutorial with me and was already ready with the installation after 10mins. 
It took him another 10mins to complete the task after understanding how Solr works and not having any error messages showing up.
